opam 2.1 integration

rjbou opened the pull request:
When opam depext is called with opam 2.1, it uses opam' internal depext mechanism (by calling opam for each features). Options are kept unchanged.
fix #141 & ocaml/opam#4854

rjbou committed:
correct opam orthograph

rjbou committed:
opam 2.1 compatibility

AltGr commented:
Thanks! Maybe worth adding a note in the 2.1 case: `You are using opam 2.1, where external dependency handling has been integrated: consider calling opam directly, the "depext" plugin interface is provided for backwards compatibility only`

rjbou committed:
add deprecation message

rjbou commented:
added!

rjbou commented on depext.ml:
```diff
@@ -90,9 +90,13 @@ let has_prefix s pfx =
     true
   with Exit -> false
 
+let is_opam_2_1 =
```
Add `--cli 2.1` for opam commands

rjbou committed:
fix depext install with pins and add cli=2.1 argument

rjbou committed:
update notice

rjbou committed:
fix and bump version to 1.2.0

dra27 committed:
Tweaks

- Use functions to propagate --cli=2.1 automatically
- Treat the branching as 2.0 vs 2.1+
- Only display the warning if OPAMCLI isn't set to 2.0 (i.e. one's using
  opam 2.1 but not requested CLI 2.0)
- Set OPAMCLI=2.0 in the environment (will be ignored by the --cli=2.1
  calls)

rjbou committed:
fix install with no package given

rjbou committed:
spaces

dra27 commented on depext.ml:
```diff
@@ -374,43 +421,83 @@ let main print_flags list short
+          let opam_install =
+            "install" :: opam_packages @ opam_install_args
+          in
+          if install_arg then opam_install else opam_install @ ["--depext-only"]
```
There's a slight issue here - `--depext-only` installs without prompting, so we need to treat this as sudo-like and have the "Allow" prompt a la `sudo_run_command`

dra27 reviewed (commented):
This seems to be working very nicely, apart from a slight regression with interactive modes

dra27 committed:
List.filter_map is 4.08+

dra27 committed:
|> is 4.01+

dra27 committed:
Sync opam file with opam-repository's

Now passes opam lint as well.

rjbou committed:
Add question when trying to install depext

dra27 reviewed (approved):
LGTM!
